WordPress/Classes/ViewRelated/Comments/Controllers/CommentComposerViewController.swift

```diff
@@ -0,0 +1,321 @@
+        textView.resignFirstResponder()
+        textView.alpha = isLoading ? 0.5 : 1.0
+        textView.isUserInteractionEnabled = !isLoading
+    }
+
+    @objc private func buttonCancelTapped() {
+        if text.isEmpty {
+            presentingViewController?.dismiss(animated: true)
+        } else {
+            showCloseDraftConfirmationAlert()
+        }
+    }
+
+    private func showCloseDraftConfirmationAlert() {
+        let alert = UIAlertController(title: nil, message: nil, preferredStyle: .actionSheet)
+        alert.addCancelActionWithTitle(Strings.closeConfirmationAlertCancel)
+        alert.addDestructiveActionWithTitle(Strings.closeConfirmationAlertDelete) { [weak self] _ in
+            self?.presentingViewController?.dismiss(animated: true)
+        }
+        // TODO: (kean) implement draft saving
+//        alert.addActionWithTitle(Strings.closeConfirmationAlertSaveDraft, style: .default) { _ in
+//
+//        }
+        alert.popoverPresentationController?.barButtonItem = navigationItem.leftBarButtonItem
+        present(alert, animated: true, completion: nil)
+    }
+
+    // MARK: - Private
+
+    private func setupNavigationBar() {
+        title = viewModel.navigationTitle
+
+        navigationItem.leftBarButtonItem = UIBarButtonItem(title: SharedStrings.Button.cancel, style: .plain, target: self, action: #selector(buttonCancelTapped))
+
+        buttonSend.heightAnchor.constraint(equalToConstant: 36).isActive = true
+        navigationItem.rightBarButtonItem = UIBarButtonItem(customView: buttonSend)
+        buttonSend.addTarget(self, action: #selector(buttonSendTapped), for: .primaryActionTriggered)
+    }
+
+    /// Changes the `refreshButton` enabled state
+    private func updateInterface() {
+        let isEmpty = text.isEmpty
+        buttonSend.isEnabled = !isEmpty
+        isModalInPresentation = !isEmpty
+    }
+
+    private var text: String {
+        textView.text.trimmingCharacters(in: .whitespacesAndNewlines)
+    }
+}
+
+extension CommentComposerViewController: UITextViewDelegate {
+    func textViewDidChange(_ textView: UITextView) {
+        placeholderLabel.isHidden = !textView.text.isEmpty
+
+        updateInterface()
+    }
+
+    func textViewDidChangeSelection(_ textView: UITextView) {
+        if didChangeText {
+            didChangeText = false
+            return
+        }
+        suggestionsView?.hideSuggestions()
+    }
+
+    func textView(_ textView: UITextView, shouldChangeTextIn range: NSRange, replacementText text: String) -> Bool {
+        guard suggestionsView != nil else {
+            return true
+        }
+        let textViewText: NSString = textView.text as NSString
+        let prerange = NSMakeRange(0, range.location)
+        let pretext = textViewText.substring(with: prerange) + text
+        let words = pretext.components(separatedBy: CharacterSet.whitespacesAndNewlines)
+        let lastWord: NSString = words.last! as NSString
```

Comment:
Unrelated to this PR. Maybe `NLTokenizer(unit: .word)` is a better approach for this. Some languages (Chinese for one) do not use space to separate words.